src/merkle_tree.cpp

```diff
@@ -89,6 +89,74 @@ namespace aux {
 		return true;
 	}
 
+	std::map<piece_index_t, std::vector<int>> merkle_tree::add_hashes(
+		int dest_start_idx
+		, int const blocks_per_piece
+		, span<sha256_hash const> tree
+		, span<std::pair<sha256_hash, sha256_hash> const> proofs)
+	{
+		std::map<piece_index_t, std::vector<int>> ret;
```

Comment:
I think this should be given a more descriptive name since the meaning of the return value is not obvious from the function signature. Something like `failed_blocks`.